apps/ui/src/components/EditorStatement.vue

```diff
@@ -0,0 +1,103 @@
+<script setup lang="ts">
+import { clone } from '@/helpers/utils';
+import { getValidator } from '@/helpers/validation';
+import type { Statement } from '@/types';
+
+const model = defineModel<Statement>({
+  required: true
+});
+
+const emit = defineEmits<{
+  (e: 'close');
+}>();
+
+const actions = useActions();
+
+const sending = ref(false);
+const previewEnabled = ref(false);
+const form = reactive(clone(model.value));
+const formErrors = ref({} as Record<string, any>);
+const formValidated = ref(false);
+
+const STATUS_DEFINITION = {
+  enum: ['ACTIVE', 'INACTIVE'],
+  options: [
+    { id: 'ACTIVE', name: 'Active' },
+    { id: 'INACTIVE', name: 'Inactive' }
+  ],
+  title: 'Status'
+};
+
+const formValidator = getValidator({
+  $async: true,
+  type: 'object',
+  title: 'Statement',
+  additionalProperties: false,
+  required: [],
+  properties: {
+    statement: {
+      type: 'string',
+      format: 'long',
+      title: 'Statement',
+      maxLength: 10000
+    },
+    status: STATUS_DEFINITION
+  }
+});
+
+async function handleSubmit() {
+  sending.value = true;
+
+  try {
+    await actions.updateStatement(form);
+    model.value = form;
+    emit('close');
+  } finally {
+    sending.value = false;
+  }
+}
+
+watchEffect(async () => {
+  formValidated.value = false;
+
+  formErrors.value = await formValidator.validateAsync({
```

Comment:
We could use non-async version there as we don't use async properties (`ens-or-address`).